workitem/workitem_repository.go

```diff
@@ -110,35 +110,31 @@ func (r *GormWorkItemRepository) Save(ctx context.Context, wi app.WorkItem) (*ap
 		return nil, errors.NewBadParameterError("Type", wi.Type)
 	}
 
-	newWi := WorkItem{
-		ID:      id,
-		Type:    wi.Type,
-		Version: wi.Version + 1,
-		Fields:  Fields{},
-	}
+	res.Version = res.Version + 1
+	res.Type = wi.Type
```

Comment:
As of now, we are allowing the type, but there is a discussion in UX to not allow. Until that is getting clarified let it be there.

Will reset `Fields`.